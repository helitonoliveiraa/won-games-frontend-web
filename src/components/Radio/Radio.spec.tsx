import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { renderWithTheme } from '@/utils/tests/helpers';

import { Radio } from './index';

describe('<Radio/>', () => {
  it('should render with label', () => {
    const { container } = renderWithTheme(
      <Radio label="radio label" labelFor="radio" />,
    );

    expect(screen.getByRole('radio')).toBeInTheDocument();
    expect(screen.getByLabelText(/radio label/i)).toBeInTheDocument();
    expect(screen.getByText(/radio label/i)).toHaveAttribute('for', 'radio');
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render without label', () => {
    renderWithTheme(<Radio value="testValue" />);

    expect(screen.getByRole('radio')).toBeInTheDocument();
    expect(screen.queryByLabelText(/radio label/i)).not.toBeInTheDocument();
  });

  it('should render black label', () => {
    renderWithTheme(
      <Radio
        label="White label"
        labelFor="radio"
        labelColor="black"
        value="testValue"
      />,
    );

    expect(screen.getByText(/white label/i)).toHaveStyle({
      color: '#030517',
    });
  });

  it('should dispatch onCheck when state change', async () => {
    const onCheck = jest.fn();

    renderWithTheme(
      <Radio
        value="testValue"
        onCheck={onCheck}
        label="Radio"
        labelFor="test radio"
      />,
    );

    expect(onCheck).not.toBeCalled();

    userEvent.click(screen.getByLabelText(/radio/i));
    await waitFor(() => {
      expect(onCheck).toHaveBeenCalledTimes(1);
    });
    expect(onCheck).toHaveBeenCalledWith('testValue');
  });

  it('should be accessible with tab', () => {
    renderWithTheme(
      <Radio value="testValue" label="Radio" labelFor="test radio" />,
    );

    expect(document.body).toHaveFocus();

    userEvent.tab();

    expect(screen.getByLabelText(/radio/i)).toHaveFocus();
  });
});
