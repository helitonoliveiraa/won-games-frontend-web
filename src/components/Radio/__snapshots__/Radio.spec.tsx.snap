// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio/> should render with label 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  width: 2rem;
  height: 2rem;
  border: 0.2rem solid #2E2F42;
  border-radius: 50%;
  background: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  position: relative;
  -webkit-transition: background 0.1s ease-in-out;
  transition: background 0.1s ease-in-out;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1::before {
  content: '';
  width: 1rem;
  height: 1rem;
  border-radius: 50%;
  background: #F231A5;
  opacity: 0;
  -webkit-transition: opacity 0.1s ease-in-out;
  transition: opacity 0.1s ease-in-out;
  position: absolute;
}

.c1:focus {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c1:checked {
  border-color: #F231A5;
}

.c1:checked::before {
  opacity: 1;
}

.c2 {
  color: #FAFAFA;
  margin-left: 0.8rem;
  line-height: 1;
}

<div
  class="c0"
>
  <input
    class="c1"
    id="radio"
    type="radio"
    value=""
  />
  <label
    class="c2"
    for="radio"
  >
    radio label
  </label>
</div>
`;
